From 64ced1632e277e4fc0b1f4de3f5bf229c6cf885b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Mon, 12 Oct 2020 14:08:28 +0200
Subject: [PATCH] window: Don't override tile monitor

Commit 033f0d11bf added a fallback in case the tile monitor wasn't
set before, but didn't actually check for a previously set value.
As a result, the "fallback" is not set unconditionally, which may
differ from the expected monitor: The tile monitor is determined
by the pointer position, while the window's monitor is the one
where the biggest part of the window resides on.

https://gitlab.gnome.org/GNOME/mutter/-/issues/1389
---
 src/core/window.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/window.c b/src/core/window.c
index dd48820819..ce6b01bfdf 100644
--- a/src/core/window.c
+++ b/src/core/window.c
@@ -3158,7 +3158,7 @@ meta_window_tile (MetaWindow   *window,
       window->tile_monitor_number = -1;
       return;
     }
-  else
+  else if (window->tile_monitor_number < 0)
     {
       window->tile_monitor_number = window->monitor->number;
     }
-- 
GitLab

